src/libostree/ostree-repo-pull.c

```diff
@@ -1498,6 +1501,114 @@ ostree_verify_unwritten_commit (OtPullData                 *pull_data,
+          g_autofree GVariantType *signature_format = NULL;
+          g_autofree gchar *pk_ascii = NULL;
+          g_autofree gchar *pk_file = NULL;
+
+          if ((sign = ostree_sign_get_by_name (names[i], error)) == NULL)
+          {
+              g_clear_error (error);
+              continue;
+          }
+          signature_key = ostree_sign_metadata_key (sign);
+          signature_format = (GVariantType *) ostree_sign_metadata_format (sign);
+
+          signatures = g_variant_lookup_value (detached_metadata,
+                                               signature_key,
+                                               signature_format);
+
+          if (!signatures)
+            continue;
+
+          /* Load keys for remote from file */
+          ostree_repo_get_remote_option (pull_data->repo,
+                                         pull_data->remote_name,
+                                         "verification-file", NULL,
+                                         &pk_file, NULL);
+          if (pk_file != NULL)
+            {
+              g_autoptr (GVariantBuilder) builder = NULL;
+              g_autoptr (GVariant) options = NULL;
+
+              builder = g_variant_builder_new (G_VARIANT_TYPE ("a{sv}"));
+              g_variant_builder_add (builder, "{sv}", "filename", g_variant_new_string (pk_file));
+              options = g_variant_builder_end (builder);
+
+              if (!ostree_sign_load_pk (sign, options, error))
+                g_clear_error (error);
+            }
+
+          /* Override key if it is set explicitly */
+          ostree_repo_get_remote_option (pull_data->repo,
+                                         pull_data->remote_name,
+                                         "verification-key", NULL,
+                                         &pk_ascii, NULL);
+          if (pk_ascii != NULL)
+            {
+              g_autoptr (GVariant) pk = NULL;
+
+              if (!g_strcmp0(ostree_sign_get_name(sign), "dummy"))
+                {
+                  // Just use the string as signature
+                  pk = g_variant_new_string(pk_ascii);
+                }
+              else if (!g_strcmp0(ostree_sign_get_name(sign), "ed25519"))
+                {
+                  gsize key_len = 0;
+                  g_autofree guchar *key = g_base64_decode (pk_ascii, &key_len);
+                  pk = g_variant_new_fixed_array (G_VARIANT_TYPE_BYTE, key, key_len, sizeof(guchar));
+                }
+
+              if (!ostree_sign_set_pk (sign, pk, error))
+                g_clear_error (error);
+            }
+
+          /* Set return to true if any sign fit */
+          if (ostree_sign_metadata_verify (sign,
+                                           signed_data,
+                                           signatures,
+                                           error
+                                          ))
+            ret = TRUE;
+          else
+            g_clear_error (error);
```

Comment:
Rather than clearing the error, you may consider accumulating the inner errors.
That could give useful informations.

Reply:
see comment above.